cpu-sim/README.md

```diff
@@ -0,0 +1,77 @@
+# CPU Simulator (`cpu-sim`)
+
+A command-line RISC-V RV32I CPU simulator that runs ELF executables on the Verilated hardware model.
+
+## Features
+
+- Loads RISC-V ELF executables
+- Simulates the single-cycle RV32I CPU with external memory
+- Supports the "tohost" mechanism for program termination (write to 0xFFFFFFF0)
+- Configurable maximum cycle limit
+- Verbose logging for debugging
+
+## Usage
+
+```bash
+# Basic usage
+cargo run --package cpu-sim -- --elf <path-to-elf-file>
+
+# With verbose logging
+cargo run --package cpu-sim -- --elf <path-to-elf-file> --verbose
+
+# Custom cycle limit
+cargo run --package cpu-sim -- --elf <path-to-elf-file> --max-cycles 50000
+
+# Or build and run directly
+cargo build --package cpu-sim
+./target/debug/cpu-sim --elf program.elf
+```
+
+## Options
+
+- `--elf <PATH>`: Path to the RISC-V ELF executable (required)
```

Comment:
This should just be a positional argument since it's required. Adding --elf every time makes usage unnecessarily verbose.